Source/msg.cpp

```diff
@@ -2487,15 +2503,15 @@ void DeltaLoadLevel()
 	}
 
 	if (leveltype != DTYPE_TOWN) {
-		for (int i = 0; i < MAXOBJECTS; i++) {
-			switch (deltaLevel.object[i].bCmd) {
+		for (auto &objectDelta : deltaLevel.object) {
+			switch (objectDelta.second.bCmd) {
 			case CMD_OPENDOOR:
 			case CMD_OPERATEOBJ:
 			case CMD_PLROPOBJ:
-				DeltaSyncOpObject(Objects[i]);
+				DeltaSyncOpObject(ObjectAtPosition(objectDelta.first));

```

Comment:
This doesn't handle malicious messages that send an invalid position. Probably need to check an object actually exists and remove the delta if nothing is on the tile.